### ISSUE TITLE
Prepared statement when single parameter is null

### ISSUE THREAD

**Issue by schveiguy** [mysql-d/mysql-native#219]
Not sure if this is a bug, but this threw me for a loop:

```D
conn.exec("INSERT INTO sometable (nullableCol) VALUES (?)", null);
```

This results in an error saying the number of parameters mismatch. But why? Because null matches a `Variant[]` before it matches the varargs version!

I think the way to fix this is to handle the `Variant[]` case specifically in the varargs template, but only if the single parameter is not `typeof(null)`.

**Comment by Abscissa** [comment]
Ouch, good catch. Sounds like this issue would be equally applicable to 
the new post-Variant API, I assume?

**Comment by schveiguy** [comment]
Yes, the new API has methods that accept `MySQLVal[]`, which would have the same problem.
